anker/src/processor.rs

```diff
@@ -236,17 +238,27 @@ fn process_sell_rewards(program_id: &Pubkey, accounts_raw: &[AccountInfo]) -> Pr
         spl_token::state::Mint::unpack_from_slice(&accounts.b_sol_mint.data.borrow())?;
     let b_sol_supply = token_mint_state.supply;
 
-    let st_sol_reserve_state = spl_token::state::Account::unpack_from_slice(
-        &accounts.st_sol_reserve_account.data.borrow(),
-    )?;
-    let reserve_st_sol = StLamports(st_sol_reserve_state.amount);
+    let reserve_st_sol = StLamports(Anker::get_token_amount(accounts.st_sol_reserve_account)?);
 
     // Get StLamports corresponding to the amount of b_sol minted.
     let st_sol_amount = solido.exchange_rate.exchange_sol(Lamports(b_sol_supply))?;
 
     // If `reserve_st_sol` < `st_sol_amount` something went wrong, and we abort the transaction.
     let rewards = (reserve_st_sol - st_sol_amount)?;
-    swap_rewards(program_id, rewards, &anker, &accounts)
+
+    // Get the amount of UST that we had.
+    let ust_before = MicroUst(Anker::get_token_amount(accounts.ust_reserve_account)?);
+    swap_rewards(program_id, rewards, &anker, &accounts)?;
+    // Get new UST amount.
+    let ust_after = MicroUst(Anker::get_token_amount(accounts.ust_reserve_account)?);
+    let swapped_ust = (ust_after - ust_before)?;
+
+    msg!("Swapped {} for {}.", st_sol_amount, swapped_ust);
```

Comment:
Nice to also print what happened. I think in hindsight one of the things I run into with the Solido program is that it is hard to see on block explorers what kind of transaction something was (aside from deposit/withdraw where we also print). Printing some hint of what we are doing fixes that. :+1: